api/scpca_portal/models/computed_file.py

```diff
@@ -126,14 +116,15 @@ def get_project_file(cls, project, download_config: Dict, computed_file_name: st
             )
 
             # Metadata file
-            output_file_constant = (
-                f'{download_config["modality"]}_METADATA_FILE_NAME'
-                if not download_config["metadata_only"]
-                else "METADATA_ONLY_FILE_NAME"
-            )
-            zip_file.write(
-                local_metadata_path, getattr(ComputedFile.MetadataFilenames, output_file_constant)
-            )
+            with io.StringIO() as metadata_buffer:
+                libraries_metadata = [
+                    lib for library in libraries for lib in library.get_combined_library_metadata()
+                ]
+                # Write libraries metadata to buffer
+                metadata_file.write_metadata_dicts(libraries_metadata, buffer=metadata_buffer)
+                output_metadata_file_name = metadata_file.get_metadata_file_name(download_config)
+                # Write buffer directly to zip archive
+                zip_file.writestr(output_metadata_file_name, metadata_buffer.getvalue())
```

Comment:
```suggestion
            libraries_metadata = [
                lib for library in libraries for lib in library.get_combined_library_metadata()
            ]

            metadata_file_name = metadata_file.get_file_name(download_config)
            metadata_file_contents = metadata_file.get_file_contents(libraries_metadata)

            # Write buffer directly to zip archive
            zip_file.writestr(output_metadata_file_name, metadata_file_contents)
```

I think this is the general flow we should be going for. The `metadata_file.get_file_contents` should handle the buffter under the hood.

Reply:
I think this makes sense. Our goal should be creating a clean interface for the caller while abstracting away implementation details, and this better accomplishes that.